modules/oe_translation_cdt/tests/src/Kernel/TranslationRequestMapperTest.php

```diff
@@ -0,0 +1,144 @@
+      'source_language_code' => 'en',
+    ];
+    $request = TranslationRequest::create($test_data);
+    assert($request instanceof TranslationRequestCdtInterface);
+    $request->updateTargetLanguageStatus('es', TranslationRequestRemoteInterface::STATUS_LANGUAGE_REQUESTED);
+    $request->updateTargetLanguageStatus('fr', TranslationRequestRemoteInterface::STATUS_LANGUAGE_REQUESTED);
+    $request->setContentEntity($entity);
+    $data = $this->container->get('oe_translation.translation_source_manager')->extractData($entity->getUntranslated());
+    $request->setData($data);
+
+    new Settings([
+      'cdt.api_key' => 'test_api_key',
+    ]);
+
+    $requestMapper = $this->container->get('oe_translation_cdt.translation_request_mapper');
+    assert($requestMapper instanceof TranslationRequestMapper);
+    $dto = $requestMapper->convertEntityToDto($request);
+    $this->assertEquals($test_data['comments'], $dto->getComments());
+    $this->assertEquals('Translation request #' . $request->id(), $dto->getTitle());
+    $this->assertEquals($request->id(), $dto->getClientReference());
+    $this->assertEquals('Translation', $dto->getService());
+    $this->assertEquals($test_data['phone_number'], $dto->getPhoneNumber());
+    $this->assertFalse($dto->isQuotationOnly());
+    $this->assertEquals('Send', $dto->getSendOptions());
+    $this->assertEquals('WS', $dto->getPurposeCode());
+    $this->assertEquals($test_data['department'], $dto->getDepartmentCode());
+    $this->assertEquals($test_data['contact_usernames'], $dto->getContactUserNames());
+    $this->assertEquals($test_data['deliver_to'], $dto->getDeliveryContactUsernames());
+    $this->assertEquals($test_data['priority'], $dto->getPriorityCode());
+
+    $source_document = $dto->getSourceDocuments()[0];
+    $this->assertEquals(['EN'], $source_document->getSourceLanguages());
+    $this->assertFalse($source_document->isPrivate());
+    $this->assertEquals('XM', $source_document->getOutputDocumentFormatCode());
+    $this->assertEquals($test_data['confidentiality'], $source_document->getConfidentialityCode());
+
+    $jobs = $source_document->getTranslationJobs();
+    foreach (['ES', 'FR'] as $key => $job_language) {
+      $this->assertEquals($job_language, $jobs[$key]->getTargetLanguage());
+      $this->assertEquals('EN', $jobs[$key]->getSourceLanguage());
+      $this->assertEquals(0.5, $jobs[$key]->getVolume());
+    }
+
+    $file = $source_document->getFile();
+    $this->assertEquals(
+      sprintf(
+        'translation_job_%s_request.xml',
+        $request->id()),
+      $file->getFileName()
+    );
+
+    $callbacks = $dto->getCallbacks();
+    foreach (['JOB_STATUS', 'REQUEST_STATUS'] as $key => $callback_type) {
+      $this->assertEquals('test_api_key', $callbacks[$key]->getClientApiKey());
+      $this->assertTrue(UrlHelper::isValid($callbacks[$key]->getCallbackBaseUrl()));
+      $this->assertEquals($callback_type, $callbacks[$key]->getCallbackType());
```

Comment:
I see a lot of positive testing, I think its not a bad idea to see some negative as well, for example what the output is if comments are not set (since they are optional), maybe check only with required fields first, then with all, and then maybe set some wrong values and check how it handles those

Reply:
There's not much to test here, but I've added checking optional parameters and expecting exception when the entity is not complete. I also decided to check the volume counting.